ccl/storageccl: TestRandomKeyAndTimestampExport failed

[(ccl/storageccl).TestRandomKeyAndTimestampExport failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=1720335&tab=buildLog) on [master@9e290a564deda51b1db8867e4d0493f3ec1d1d96](https://github.com/cockroachdb/cockroach/commits/9e290a564deda51b1db8867e4d0493f3ec1d1d96):

```
=== RUN   TestRandomKeyAndTimestampExport
I200131 06:22:48.719816 1901 storage/engine/rocksdb.go:603  opening rocksdb instance at "/tmp/TestRandomKeyAndTimestampExport922116276"
--- FAIL: TestRandomKeyAndTimestampExport (414.35s)
=== RUN   TestRandomKeyAndTimestampExport/targetSize=65536/kv_(randLowerTime,_randUpperTime],_all,_nontimebound
        --- FAIL: TestRandomKeyAndTimestampExport/targetSize=65536/kv_(randLowerTime,_randUpperTime],_all,_nontimebound (0.96s)
            require.go:1104: 
                	Error Trace:	export_test.go:374
                	Error:      	Expect "<nil>" to match "export size \(65536 bytes\) exceeds max size \(65536 bytes\)"
                	Test:       	TestRandomKeyAndTimestampExport/targetSize=65536/kv_(randLowerTime,_randUpperTime],_all,_nontimebound
=== RUN   TestRandomKeyAndTimestampExport/targetSize=65536/kv_(randLowerTime,_randUpperTime],_all,_timebound
        --- FAIL: TestRandomKeyAndTimestampExport/targetSize=65536/kv_(randLowerTime,_randUpperTime],_all,_timebound (0.92s)
            require.go:1104: 
                	Error Trace:	export_test.go:374
                	Error:      	Expect "<nil>" to match "export size \(65536 bytes\) exceeds max size \(65536 bytes\)"
                	Test:       	TestRandomKeyAndTimestampExport/targetSize=65536/kv_(randLowerTime,_randUpperTime],_all,_timebound
=== RUN   TestRandomKeyAndTimestampExport/targetSize=65536
    --- FAIL: TestRandomKeyAndTimestampExport/targetSize=65536 (9.59s)

```

<details><summary>More</summary><p>
Parameters:

- GOFLAGS=-json

```
make stressrace TESTS=TestRandomKeyAndTimestampExport PKG=./pkg/ccl/storageccl TESTTIMEOUT=5m STRESSFLAGS='-timeout 5m' 2>&1
```


[See this test on roachdash](https://roachdash.crdb.dev/?filter=status%3Aopen+t%3A.%2ATestRandomKeyAndTimestampExport.%2A&sort=title&restgroup=false&display=lastcommented+project)
<sub>powered by [pkg/cmd/internal/issues](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)</sub></p></details>

[(ccl/storageccl).TestRandomKeyAndTimestampExport failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=1722772&tab=buildLog) on [master@1347bd4cba1031e6bb7d964be2d5066a19b8df20](https://github.com/cockroachdb/cockroach/commits/1347bd4cba1031e6bb7d964be2d5066a19b8df20):

```
=== RUN   TestRandomKeyAndTimestampExport
I200202 04:23:18.081867 1995 storage/engine/rocksdb.go:603  opening rocksdb instance at "/tmp/TestRandomKeyAndTimestampExport604273049"
--- FAIL: TestRandomKeyAndTimestampExport (227.97s)
=== RUN   TestRandomKeyAndTimestampExport/targetSize=65536
    --- FAIL: TestRandomKeyAndTimestampExport/targetSize=65536 (10.35s)
=== RUN   TestRandomKeyAndTimestampExport/targetSize=65536/kv_[randLower,_randUpper),_latest,_nontimebound
        --- FAIL: TestRandomKeyAndTimestampExport/targetSize=65536/kv_[randLower,_randUpper),_latest,_nontimebound (0.22s)
            require.go:1104: 
                	Error Trace:	export_test.go:374
                	Error:      	Expect "<nil>" to match "export size \(65536 bytes\) exceeds max size \(65536 bytes\)"
                	Test:       	TestRandomKeyAndTimestampExport/targetSize=65536/kv_[randLower,_randUpper),_latest,_nontimebound
=== RUN   TestRandomKeyAndTimestampExport/targetSize=65536/kv_[randLower,_randUpper),_latest,_timebound
        --- FAIL: TestRandomKeyAndTimestampExport/targetSize=65536/kv_[randLower,_randUpper),_latest,_timebound (0.20s)
            require.go:1104: 
                	Error Trace:	export_test.go:374
                	Error:      	Expect "<nil>" to match "export size \(65536 bytes\) exceeds max size \(65536 bytes\)"
                	Test:       	TestRandomKeyAndTimestampExport/targetSize=65536/kv_[randLower,_randUpper),_latest,_timebound

```

<details><summary>More</summary><p>
Parameters:

- GOFLAGS=-json

```
make stressrace TESTS=TestRandomKeyAndTimestampExport PKG=./pkg/ccl/storageccl TESTTIMEOUT=5m STRESSFLAGS='-timeout 5m' 2>&1
```


[See this test on roachdash](https://roachdash.crdb.dev/?filter=status%3Aopen+t%3A.%2ATestRandomKeyAndTimestampExport.%2A&sort=title&restgroup=false&display=lastcommented+project)
<sub>powered by [pkg/cmd/internal/issues](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)</sub></p></details>

[(ccl/storageccl).TestRandomKeyAndTimestampExport failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=1724773&tab=buildLog) on [master@5337d69d22c73a2316828c3c5ad6dc90736cb83e](https://github.com/cockroachdb/cockroach/commits/5337d69d22c73a2316828c3c5ad6dc90736cb83e):

```
=== RUN   TestRandomKeyAndTimestampExport
I200203 12:58:50.325281 2101 storage/engine/rocksdb.go:603  opening rocksdb instance at "/tmp/TestRandomKeyAndTimestampExport846052294"
--- FAIL: TestRandomKeyAndTimestampExport (369.02s)
=== RUN   TestRandomKeyAndTimestampExport/targetSize=65536
    --- FAIL: TestRandomKeyAndTimestampExport/targetSize=65536 (11.31s)
=== RUN   TestRandomKeyAndTimestampExport/targetSize=65536/kv_(randLowerTime,_randUpperTime],_all,_nontimebound
        --- FAIL: TestRandomKeyAndTimestampExport/targetSize=65536/kv_(randLowerTime,_randUpperTime],_all,_nontimebound (0.68s)
            require.go:1104: 
                	Error Trace:	export_test.go:374
                	Error:      	Expect "export size (65845 bytes) exceeds max size (65536 bytes)" to match "export size \(65536 bytes\) exceeds max size \(65536 bytes\)"
                	Test:       	TestRandomKeyAndTimestampExport/targetSize=65536/kv_(randLowerTime,_randUpperTime],_all,_nontimebound
=== RUN   TestRandomKeyAndTimestampExport/targetSize=65536/kv_(randLowerTime,_randUpperTime],_all,_timebound
        --- FAIL: TestRandomKeyAndTimestampExport/targetSize=65536/kv_(randLowerTime,_randUpperTime],_all,_timebound (0.76s)
            require.go:1104: 
                	Error Trace:	export_test.go:374
                	Error:      	Expect "export size (65845 bytes) exceeds max size (65536 bytes)" to match "export size \(65536 bytes\) exceeds max size \(65536 bytes\)"
                	Test:       	TestRandomKeyAndTimestampExport/targetSize=65536/kv_(randLowerTime,_randUpperTime],_all,_timebound

```

<details><summary>More</summary><p>
Parameters:

- GOFLAGS=-json

```
make stressrace TESTS=TestRandomKeyAndTimestampExport PKG=./pkg/ccl/storageccl TESTTIMEOUT=5m STRESSFLAGS='-timeout 5m' 2>&1
```


[See this test on roachdash](https://roachdash.crdb.dev/?filter=status%3Aopen+t%3A.%2ATestRandomKeyAndTimestampExport.%2A&sort=title&restgroup=false&display=lastcommented+project)
<sub>powered by [pkg/cmd/internal/issues](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)</sub></p></details>

--- a/pkg/ccl/storageccl/export_test.go
+++ b/pkg/ccl/storageccl/export_test.go
@@ -365,6 +365,12 @@ func assertEqualKVs(
 					lessThisKey := dataSizeWhenExceeded - uint64(len(kv.Key.Key)+len(kv.Value))
 					if lessThisKey >= maxSize {
 						dataSizeWhenExceeded = lessThisKey
+						// It might be the case that this key would lead to an SST of exactly
+						// max size, in this case we overwrite max size to be less so that
+						// we still generate an error.
+						if dataSizeWhenExceeded == maxSize {
+							maxSize--
+						}
 					} else {
 						break
 					}

